pkg/mlsstore/store.go

```diff
@@ -2,27 +2,170 @@ package mlsstore
 
 import (
 	"context"
+	"crypto/sha256"
+	"database/sql"
+	"encoding/hex"
+	"errors"
+	"time"
 
 	"github.com/uptrace/bun"
+	"github.com/uptrace/bun/migrate"
+	"github.com/xmtp/xmtp-node-go/pkg/migrations/messages"
 	"go.uber.org/zap"
 )
 
 type Store struct {
-	ctx    context.Context
-	cancel context.CancelFunc
+	config Config
 	log    *zap.Logger
 	db     *bun.DB
 }
 
 type MlsStore interface {
+	CreateInstallation(ctx context.Context, installationId string, walletAddress string, lastResortKeyPackage []byte) error
+	InsertKeyPackages(ctx context.Context, keyPackages []*KeyPackage) error
+	ConsumeKeyPackages(ctx context.Context, installationIds []string) ([]*KeyPackage, error)
 }
 
-func New(config Config) (*Store, error) {
+func New(ctx context.Context, config Config) (*Store, error) {
 	s := &Store{
-		log: config.Log.Named("mlsstore"),
-		db:  config.DB,
+		log:    config.Log.Named("mlsstore"),
+		db:     config.DB,
+		config: config,
+	}
+
+	if err := s.migrate(ctx); err != nil {
+		return nil, err
 	}
-	s.ctx, s.cancel = context.WithCancel(context.Background())
 
 	return s, nil
 }
+
+func (s *Store) Close() {
+	if s.db != nil {
+		s.db.Close()
```

Comment:
Since `s.db` is instantiated outside and passed in via `Config`, it's probably on the caller to close it down too rather than closing it down in here.

Reply:
I came to the same conclusion and fixed that in the next PR